auth/genericoauth/provider.go

```diff
@@ -141,9 +142,12 @@ func (GenericTeamProvider) ProviderConstructor(
 	}
 
 	var oauthVerifier verifier.Verifier
-	if genericOAuth.Scope != "" {
+	if genericOAuth.Scope != "" && genericOAuth.Authority = "" {
```

Comment:
This will result in configuring no verification at all if both are configured, which seems risky or at least counterintuitive.

Does it make sense to check for both a scope and an authority? Or should that be a validation error? Or should one take precedence?